godotenv.go

```diff
@@ -323,7 +323,10 @@ func expandVariables(v string, m map[string]string) string {
 		if submatch[1] == "\\" || submatch[2] == "(" {
 			return submatch[0][1:]
 		} else if submatch[4] != "" {
-			return m[submatch[4]]
+			if val, ok := m[submatch[4]]; ok {
+				return val
+			}
+			return os.Getenv(submatch[4])
```

Comment:
```suggestion
			if val, ok := os.LookupEnv(submatch[4]); ok {
				return val
			}
```

nit: this way if the variable is not defined yet it will remain unexpanded allowing that case to be handeld by the code using this lib.